aten/src/ATen/native/cpu/FlashAttentionKernel.cpp

```diff
@@ -31,21 +33,39 @@ inline void _scale_attn_mask_fusion_kernel(
     T1& val) {
   const auto vec_size1 = at::vec::Vectorized<T1>::size();
   const auto vec_size2 = at::vec::Vectorized<T2>::size();
-  constexpr int64_t T1_n = (vec_size2 == vec_size1 * 2 && is_reduced_floating_point_v<T2>) ? 2 : 1;
+  constexpr int64_t T1_n =
+      (vec_size2 == vec_size1 * 2 && is_reduced_floating_point_v<T2>) ? 2 : 1;
   constexpr int64_t T2_n = 1;
   auto vec_scale = at::vec::VectorizedN<T1, T1_n>(val);
   int64_t i = 0;
-  for (; i < size - (size % vec_size2); i += vec_size2) {
-    auto a_n = at::vec::VectorizedN<T1, T1_n>::loadu(a + i);
-    auto b_n = at::vec::VectorizedN<T2, T2_n>::loadu(b + i);
-    auto b_n_convert = at::vec::convert<T1, T1_n, T2, T2_n, true>(b_n);
-    auto res = a_n * vec_scale + b_n_convert;
-    res.store(out + i);
-  }
-  for (; i < size; i++) {
-    auto tmp0 = a[i];
-    auto tmp1 = (T1) b[i];
-    out[i] = tmp0 * val + tmp1;
+  if (is_b_stride_zero) {
```

Comment:
I would reuse the original for loop and only insert `if constexpr(is_b_stride_zero)` where necessary in the code.

Reply:
Thanks for the advice! Applied.